### ISSUE TITLE
Sentinel without AUTH and replicas with AUTH

### ISSUE THREAD

**Issue by 1thew** [predis/predis#783]
**Describe the bug**
I'm trying to connect sentinel
Sentinel does not have authentication, but replicas do have a password.
Can't create client that uses password only for Redis replica

**To Reproduce**
```
        $sentinels = ["tcp://my_sentinel:26379"];

        $client = new Client($sentinels, [
            "replication" => "sentinel",
            "service" => "test",
            "parameters" => [
                "password" => "mypassword",
                "database" => "0",
            ],
        ]);

        $client->connect();
```
If config have  "password"  -- get an error "No sentinel server available for autodiscovery". Client send AUTH to sentinel connections.
If config dont have "password" :
`Predis\Connection\ConnectionException : `SELECT` failed: NOAUTH Authentication required. [tcp://172.16.101.3:6379]
`
Client dont use AUTH for replica.
.

**Versions (please complete the following information):**
 - Predis: 2.
 - PHP 8.1
 - Redis Server  6.2.
 - Ubuntu

Maybe I'm doing something wrong?

**Comment by tillkruss** [comment]
It's a known issue. Pull Requests welcome. See #782.

**Comment by dakorpar** [comment]
If there's no password on sentinels then you need to set password in url as empty string, otherwise it will use default redis password for sentinels and fail. This is breaking change from 1.1.10 to 2.0.0, try something like this:
```
$sentinels = array(
    'tcp://sentinel1:26379?password=',
    'tcp://sentinel2:26379?password=',
    'tcp://sentinel3:26379?password=',
);

$client = new Predis\Client($sentinels, array(
    'replication' => 'sentinel',
    'service' => 'mymaster',
    'parameters' => ['database'=> 0, 'password' => 'redis-password']
));
```

**Comment by tillkruss** [comment]
I don't have the bandwidth to test Sentinel setups. I'm open to PRs with test cases and detailed explanations.

**Comment by waqas-badar** [comment]
I was facing this issue. My issue got resolved after increasing the timeout.

I face another issue, while investigating, it is found that in SentinelReplication.php we are using SENTINEL slaves command which is not supported in new version IMHO. I tried with sentinel help command in redis-cli and unable to find it. When i switch the command to SENTINELS then it work. Not sure either it was the exact fix or not.

**Comment by tillkruss** [comment]
The missing command was re-added in 7.0.1: https://github.com/redis/redis/pull/10723

**Comment by mta59066** [comment]
Looks like this was mentioned here https://github.com/predis/predis/issues/658 and already fixed here https://github.com/predis/predis/commit/16957f3b39679b42b08ed2100debf4af89170fa4 but was reverted and broken again here https://github.com/predis/predis/commit/619890f7154ccb6aca758fe23fd4a5fd2f728caa. That commit was done by you @tillkruss do you remember what `fix test` was for?

**Comment by mta59066** [comment]
Digging further, it looks like it was broken on 1.1.5 and fixed on 1.1.6 (https://github.com/predis/predis/commit/16957f3b39679b42b08ed2100debf4af89170fa4) but 2.0.0-beta1 was merged from 1.1.5 (https://github.com/predis/predis/commit/d954e9f43f75d33a04b15a210a801cb1fe7837c8) so current code has what 1.1.6 claimed to be a bug in it. And current fixes are not merged because they don't have the appropriate tests. Was 2.0.x forked from 1.1.5, was there a reason why later developments where dropped, I don't know the history?

I'll see what I can come up with in therms of a mr. I'd appreciate any pointers.

**Comment by mta59066** [comment]
@tillkruss is there a sentinel authentication scenario currently in predis? From what I can see any authentication with sentinal just does not work and because of this issue redis authentication with sentinal does not work. As mentioned before the code committed on 1.1.5 was [admittedly](https://github.com/predis/predis/releases/tag/v1.1.6) broken.

I'll look at it but `with tests that the removal won't break other sentinel authentication scenarios` requires fully developing sentinel authentication as it does not exist in predis.

I might be wrong of course, and if there is currently working sentinel authentication implementation in the code that I am not able to see, then I am sorry and it is perfectly fine to tell everybody to use authentication, don't use an open system. But if I am correct than the current stance forces sentinel use to opening redis itself.

**Comment by tillkruss** [comment]
@mta59066: I don't know TBH, I don't use Redis Sentinel myself.

**Comment by mta59066** [comment]
I built test servers with redis-3.2.12 and redis-7.0.5, and created master-slave and multiple sentinel configurations, with and without authentication (basic authentication not user/acl). Predis sentinel authentication under these setups works fine. The way to get replica-authentication with no-sentinel-authentication is to pass empty password on the uri ('tcp://sentinel1:26379?password=') as it was suggested earlier by @dakorpar. 

The requirement for empty password is not documented and I think it should work without it, so I'll send a pr that handles that with tests that make sure current empty password method works as expected as before.

**Comment by mta59066** [comment]
pr #807

**Comment by tillkruss** [comment]
Closed via #807.
